activerecord/test/cases/insert_all_test.rb

```diff
@@ -104,6 +104,20 @@ def test_insert_all_returns_ActiveRecord_Result
     assert_kind_of ActiveRecord::Result, result
   end
 
+  def test_insert_all_validates_consistent_attributes
+    assert_raises ArgumentError do
+      Book.insert_all [{ name: "Rework" }, {}]
+    end
+
+    assert_nothing_raised do
+      Book.insert_all [{ created_at: Time.current }, {}]
+    end
```

Comment:
I think it would be cool if we said "either use record_timestamps or specify your own timestamp attributes but not both", but that would be a larger, breaking change